src/main/java/de/rub/nds/modifiablevariable/util/ArrayConverter.java

```diff
@@ -422,4 +413,26 @@ public static byte[] reverseByteOrder(byte[] array) {
         }
         return temp;
     }
+
+    /**
+     * Returns the starting index of the innerArray inside the outerArray if present. Returns -1 if
+     * the innerArray is not present in the outerArray
+     *
+     * @param outerArray Outer byte array to search for inner array
+     * @param innerArray byte array searched for
+     * @return StartIndex of innerArray in outerArray or -1 if not present.
+     */
+    public static int indexOf(byte[] outerArray, byte[] innerArray) {
+        for (int i = 0; i < outerArray.length - innerArray.length + 1; ++i) {
+            boolean found = true;
+            for (int j = 0; j < innerArray.length; ++j) {
+                if (outerArray[i + j] != innerArray[j]) {
+                    found = false;
+                    break;
+                }
+            }
+            if (found) return i;
+        }
+        return -1;
```

Comment:
Java provides `Collections.indexOfSubList()`. However, for this, we need to cast the arrays into lists.
For me, both solutions are ok.